### PR TITLE
Add GitHub Actions Build

### PR DESCRIPTION
**Pull request by stephengroat**
Note: This does change the `Debug` scheme to use locally signed certs instead of the team cert. This shouldn't affect release builds

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (reitermarkus) [inline .github/workflows/main.yml:7]: I think this only runs when the pull request head branch is `master`, which is why it isn't working for this pull request.

Comment (stephengroat) [reply .github/workflows/main.yml:7]: works on my fork pr https://github.com/stephengroat/MonitorControl/pull/1, maybe it has to be merged in?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+jobs:
+  build:
+    runs-on: macos-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
```
Comment (reitermarkus) [inline .github/workflows/main.yml:14]: Remove this comment (or replace it with a `name` for the step).

Comment (stephengroat) [reply .github/workflows/main.yml]: removed in bb84fb0cafeb27191c29c340d94b300b68fe9192

File: .github/workflows/main.yml (hunk 1 above)
Comment (JoniVR) [reply .github/workflows/main.yml:7]: Yes I think that this should work, CI should run when changes are merged into master and for each PR. It's probably indeed due to needing to be merged in first.

**Review by JoniVR — APPROVED** [review]

**Review by JoniVR — CHANGES_REQUESTED** [review]
I opened up a test PR on your branch, seems like it fails the build now throwing a linking error for Swiftlint. 

My best guess is it's probably the `brew link` step that can go since SwiftLint seems to be included by default on macOS for Github Actions? Not sure.

**Comment by stephengroat** [comment]
@JoniVR unfortunately, `Preferences` now requires `swift-tools-version` `5.4` which is xcode `12.5`, only available on `macos-11`.

I've asked for access to the private preview, I'll have to see what happens

**Comment by JoniVR** [comment]
Hi, thanks for your effort, it's much appreciated, I did some reading and found the following:

* [They no longer seem to be accepting new users/orgs for the private preview](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).

* [Xcode 12.5 on Big Sur support is supposed to launch today](https://github.com/actions/virtual-environments/issues/3522) 🙂 

They state it might take 2-3 days to propagate to that would hopefully solve it then.

**Comment by AlenaSviridenko** [comment]
Hi @JoniVR,
[VE maintainer here](https://github.com/actions/virtual-environments/issues/2486), sorry for pinging you here, but I saw several requests for this org to join macOS 11 pool, and we definitely can do that, but it requires additional action from your side. 
I have sent a couple of clarifying emails back in June and now, but they might have land to spam folder. Please, check them and after that I will be able to provide access to the macOS 11 pool.
Thanks.

**Comment by JoniVR** [comment]
Hi, thanks for pinging me here. So our best bet seems to be just creating a workflow (any workflow) that triggers Github Actions, after that it should work if I read everything correctly?

**Comment by AlenaSviridenko** [comment]
Correct, you can try it with `macos-latest`, even if it fails, it will still trigger the process

**Comment by AlenaSviridenko** [comment]
Yes, confirming that I can find the required key now, will provide access today. Thanks!

**Comment by JoniVR** [comment]
Thanks for the help, much appreciated!

**Comment by AlenaSviridenko** [comment]
macOS 11 should be accessible for you now, please check, and let me know if it doesn't work 🙂

**Review by JoniVR — APPROVED** [review]

**Comment by JoniVR** [comment]
Thanks for the PR @stephengroat!

**Pull request merged**